src/bplus_tree.rs

```diff
@@ -0,0 +1,470 @@
+use chunkfs::{Data, DataContainer, Database};
+
+use std::{
+    cell::RefCell,
+    fmt::Debug,
+    fs::File,
+    io::{self, ErrorKind},
+    os::unix::fs::FileExt,
+    path::PathBuf,
+    rc::Rc,
+};
+
+extern crate chunkfs;
+
+/// Structure that handles chunks written in files
+#[derive(Clone, Default, Debug)]
+pub struct ChunkHandler {
+    path: PathBuf,
+    offset: u64,
+    size: usize,
+}
+
+impl ChunkHandler {
+    fn new(path: PathBuf, offset: u64, size: usize) -> Self {
+        ChunkHandler { path, offset, size }
+    }
+
+    fn read(&self) -> io::Result<Vec<u8>> {
+        let file = File::open(self.path.clone()).expect("file error");
+        let mut buf = vec![0; self.size];
+        let result = file.read_exact_at(&mut buf, self.offset);
+        match result {
+            Ok(_x) => Ok(buf),
+            Err(error) => Err(error),
+        }
+    }
+}
+
+/// A type that represents a reference to another node
+type Link<K> = Rc<RefCell<Node<K>>>;
+
+/// Represents a node in a B+ tree.
+/// All data resides in leaf nodes, while internal nodes
+/// manage navigation between children
+#[derive(Clone)]
+enum Node<K> {
+    Internal(InternalNode<K>),
+    Leaf(Leaf<K>),
+}
+
+/// Internal node in a B+ tree
+#[derive(Clone)]
+struct InternalNode<K> {
+    children: Vec<Link<K>>,
+    keys: Vec<Rc<K>>,
+}
+
+/// Leaf node in a B+ tree
+#[derive(Default, Clone)]
+struct Leaf<K> {
+    entries: Vec<(Rc<K>, ChunkHandler)>,
+    next: Option<Link<K>>,
+}
+
+/// B+ tree
+pub struct BPlus<K> {
+    root: Node<K>,
+    t: usize,
+    path: PathBuf,
+    file_number: usize,
+    offset: u64,
+    current_file: File,
+    max_file_size: u64,
+}
+
+#[allow(dead_code)]
+impl<K: Ord + Debug> BPlus<K> {
+    /// Prints B+ tree for debug purposes
+    pub fn print_tree(&self) {
+        BPlus::print_node(&self.root, 0);
+    }
+
+    fn print_node(node: &Node<K>, level: usize) {
+        match node {
+            Node::Internal(internal) => {
+                println!("{}[Internal] keys: {:?}", "  ".repeat(level), internal.keys);
+                for child in &internal.children {
+                    BPlus::print_node(&child.borrow(), level + 1);
+                }
+            }
+            Node::Leaf(leaf) => {
+                let entries: Vec<String> = leaf
+                    .entries
+                    .iter()
+                    .map(|(k, _)| format!("{:?}", k))
+                    .collect();
+                println!("{}[Leaf] entries: {:?}", "  ".repeat(level), entries);
```

Comment:
Isn't it the same as.
```suggestion
                println!("{}[Leaf] entries: {:?}", "  ".repeat(level), leaf.entries);
```